### PR TITLE
Spreadsheet: Add actions to allow quick change of cell styles

### PR DESCRIPTION
**Pull request by lazynessmind**
## Brief explanation on why this came about:

While trying the Serenity `Spreadsheet` app, I found out that we couldn't change the background or foreground color without mouse-clicking 7 times.

1. Right-Click the Cell
2. Left-Click "Type and Formatting"
3. Left-Click the "Color" tab
4. Left-Click the rectangle to spawn the Color Picker Dialog on background or foreground input fields
5. Left-Click the color
6. Left-Click the "OK" button
7. Left-Click again on the "OK" button

## The "Fix"

To make it easier for the end-user, I added two new toolbar actions to decrease the number of clicks to only 3.

This is done by replacing the color of all selected cells with the color returned by the `Color Picker`.

### Changes:

- New Icon to represent text color;
- Two new actions:
  - Change Background (Ctrl+P);
  - Change Text Color (Cttl+Shift+P);

### Notes:

- Currently, the foreground color is treated as "Text Color", in my opinion, it's easier for the end-user, but causes inconsistencies.
- If the user presses the "OK" button on the `Color Picker` dialog, without choosing a color, it defaults to `white`, this is the default behavior but can be changed in the future to keep the current cell color.
- I tried my best to follow the contribution guideline and hope that I have done it correctly.

## Preview:

https://user-images.githubusercontent.com/52864251/156534680-106419b6-052c-4028-a335-3bb515f3647a.mp4

### CODE REVIEW

File: Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp [hunk 1]
```diff
@@ -429,6 +449,24 @@ void SpreadsheetWidget::redo()
     update();
 }
 
+void SpreadsheetWidget::change_cell_static_color_format(bool background)
+{
+    VERIFY(m_selected_view);
+    auto* sheet_ptr = m_selected_view->sheet_if_available();
+    VERIFY(sheet_ptr);
+    auto& sheet = *sheet_ptr;
```
Comment (kleinesfilmroellchen) [inline Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp:457]: This is kind of clunky, you should be able to do that in just one line (including the VERIFY).

File: Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp [hunk 2]
```diff
@@ -429,6 +449,24 @@ void SpreadsheetWidget::redo()
     update();
 }
 
+void SpreadsheetWidget::change_cell_static_color_format(bool background)
```
Comment (kleinesfilmroellchen) [inline Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp:452]: What does "static" mean here?

Comment (kleinesfilmroellchen) [reply Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp:452]: Also, don't use a boolean for these kinds of two-value flags, use a special enum instead. See the style guidelines for more.

File: Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp [hunk 3]
```diff
@@ -429,6 +449,24 @@ void SpreadsheetWidget::redo()
...
+{
+    VERIFY(m_selected_view);
+    auto* sheet_ptr = m_selected_view->sheet_if_available();
+    VERIFY(sheet_ptr);
+    auto& sheet = *sheet_ptr;
+
+    auto dialog = GUI::ColorPicker::construct(Color::White, window(), "Select Color");
+    if (dialog->exec() == GUI::Dialog::ExecOK) {
+        for (const auto& position : sheet.selected_cells()) {
```
Comment (kleinesfilmroellchen) [inline Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp:461]: East const

**Review by kleinesfilmroellchen — CHANGES_REQUESTED** [review]
Welcome to Serenity! This looks pretty good, just some style stuff.

File: Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp (hunk 2 above)
Comment (lazynessmind) [reply Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp:452]: > What does "static" mean here?

The static means static format is cells that don't change their style depending on the content, like if a value is less than ten turn red. That's why is called `static_color_format`. Should I change it?

Comment (lazynessmind) [reply Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp:452]: > Also, don't use a boolean for these kinds of two-value flags, use a special enum instead. See the style guidelines for more.

Something like:
```c++
enum FormatType {
    Background = 0,
    Foreground = 1
}
```

Should I use `VERIFY_NOT_REACHED()` in this case?

File: Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp (hunk 1 above)
Comment (lazynessmind) [reply Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp]: Done in f5101cf

File: Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp [hunk 4]
```diff
@@ -213,6 +214,18 @@ SpreadsheetWidget::SpreadsheetWidget(GUI::Window& parent_window, NonnullRefPtrVe
...
     });
 
+    m_change_background_color_action = GUI::Action::create(
+        "&Change Background Color", { Mod_Ctrl, Key_P }, Gfx::Bitmap::try_load_from_file("/res/icons/pixelpaint/bucket.png").release_value_but_fixme_should_propagate_errors(), [&](auto&) {
+            change_cell_static_color_format(true);
+        },
+        window());
+
+    m_change_text_color_action = GUI::Action::create(
+        "&Change Text Color", { Mod_Ctrl | Mod_Shift, Key_P }, Gfx::Bitmap::try_load_from_file("/res/icons/16x16/text-color.png").release_value_but_fixme_should_propagate_errors(), [&](auto&) {
```
Comment (alimpfard) [inline Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp:224]: These two actions share a shortcut, perhaps make it B and T instead?

**Review by alimpfard — COMMENTED** [review]
First commit typo: represend => represent.
Also, please squash the fixups into the commits they fixup.

(Overall, LGTM)

**Comment by alimpfard** [comment]
> First commit typo: represend => represent.
Also, please squash the fixups into the commits they fixup.

Additionally, this now has a conflict.

**Comment by lazynessmind** [comment]
> > First commit typo: represend => represent.
> > Also, please squash the fixups into the commits they fixup.
> 
> Additionally, this now has a conflict.

OMG pre-commit is killing me 👎🏼 
Long story short, tried to use the build-in Github tool to resolve the conflicts but the same tools push a new commit that is _lint correct_. Then I had to amend the commit and force push it, but for some reason, the pre-commit didn't work. So I had to clone the repo again and reinstall the hooks.

Sorry for wasting your time and resources on this kinda things.

**Review by kleinesfilmroellchen — CHANGES_REQUESTED** [review]
If you still want this to be merged, please put the conflict resolution in the original commits and squash the comment-addressing commits into the original commits as well. If I see this correctly, there should be three commits: One for the icon, one for the style actions, and one for the bool->enum refactor.

**Comment by lazynessmind** [comment]
I think I'm going to pass this one for someone else since I'm not in the mood right now. Sorry for the inconvenience.

**Pull request closed without merging**